src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { get5First, getAll, getRedGoods } from './api/goods';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [allGoods, setAllGoods] = useState(false);
+  const [fiveGoods, setFiveGoods] = useState(false);
+  const [redGoods, setRedGoods] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    getRedGoods().then(setGoods);
+  }, [redGoods]);
```

Comment:
```suggestion
  useEffect(() => {
    getRedGoods().then(setGoods);
  }, [redGoods]);
```
```suggestion
  const getGoods = (callback) => () => callback().then(setGoods)
```
should render any goods on initial render.
can be one handler factory which except get goods function and return arrow function
remove other useEffects